src/main/java/org/benetech/servicenet/service/impl/GeocodingResultServiceImpl.java

```diff
@@ -58,4 +72,30 @@ public void delete(UUID id) {
         log.debug("Request to delete GeocodingResult : {}", id);
         geocodingResultRepository.deleteById(id);
     }
+
+    @Override
+    public List<GeocodingResult> createOrUpdateGeocodingResult(Location location) {
+        if (location.getPhysicalAddress() == null) {
+            return new ArrayList<>();
+        }
+
+        String addressString = geoApi.extractAddressString(location.getPhysicalAddress());
+        List<GeocodingResult> currentResults = geocodingResultRepository.findAllByAddress(addressString);
```

Comment:
I recommend adding an index on that. Since it is a clob you should narrow an indexed part of an address to for example 50 chars. I think that coverage of address.getAddress1(), address.getCity() would be enough to narrow search results. loner string would be just too much computation.

Reply:
Sure, nice catch

Reply:
Unfortunatelly, as the address is of type CLOB, neither h2 or psql does not support it fully. We can create a walkaround later if it will be too slow. @wojtekbu what do you think?

Reply:
Works for me